vllm/v1/core/scheduler.py

```diff
@@ -455,30 +463,32 @@ def update_from_output(
                 # TODO: Update the KV cache manager for prefix caching.
 
                 # Check for stop and update request state.
-                # This must be called before we make the EngineCoreOutput.
+                # This must be called before we make the EngineCoreOutputs.
                 stopped = self._check_stop(request)
                 if stopped:
                     self._free_request(request)
 
-                # Add EngineCoreOutput for this Request.
-                output = EngineCoreOutput(
-                    request_id=req_id,
-                    new_token_ids=request.output_token_ids[-num_new_tokens:],
-                    finished=request.is_finished(),
-                    finish_reason=request.get_finished_reason(),
-                    stop_reason=request.stop_reason)
-                outputs.append(output)
+                # not a list of outputs here
+
+                # Add EngineCoreOutputs for this Request.
+                output.request_ids.append(req_id)
+                output.new_token_id_offsets.append(offset)
+                output.new_token_ids += request.output_token_ids[-num_new_tokens:]
+                output.finished.append(request.is_finished())
+                if request.get_finished_reason() is not None:
+                    output.finish_reason[req_id] = request.get_finished_reason()
+                #print(f"req stop = {request.stop_reason}, {request.status}")
+                output.stop_reason.append(request.stop_reason)
+                offset = offset + 1  # move out of if?
```

Comment:
```suggestion
                offset = offset + num_new_tokens
```

But could just be changed to counts per other comment